app/models/concerns/poll_status_machine.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+module PollStatusMachine
+  require 'micromachine'
+
+  extend ActiveSupport::Concern
+
+  def configure_machine(initial_state)
+    machine = PollMicroMachine.new(initial_state) # new state-machine initialized by poll status
+
+    add_callbacks(machine)
+
+    machine
+  end
+
+  private
+
+  def add_callbacks(machine)
+    machine.on('ready') { check_options_presists(machine) }
+    machine.on('draft') { drop_votation_progress }
+  end
+
+  def check_options_presists(machine)
+    if options.empty?
+      errors.add(:base, "Status can't be ready when poll have no options")
+      machine.trigger(:draft)
+    end
+  end
+
+  # Poll class methods
+  module ClassMethods
+    def availible_status_transitions(rules)
+      generate_status_machine_class(rules)
+      define_status_methods_from(rules)
+    end
+
+    private
+
+    def generate_status_machine_class(rules) # generate class for status machine with included list of available transitions
+      PollStatusMachine.const_set(:PollMicroMachine, Class.new(MicroMachine))
+      PollMicroMachine.send :define_method, :initialize do |status|
+        super(status)
+        @transitions_for = rules
+      end
+    end
+
+    def define_status_methods_from(rules)
+      statuses = rules.values.map(&:to_a).flatten.uniq
+      rules.each_key do |k| # dynamic generation methods for check and change status
+        define_change_status_method(k)
+        define_check_availability_to_change_status_method(k)
+      end
+      statuses.each do |k|
+        define_check_status_method(k)
+        define_scope_by_status_method(k)
+      end
+    end
+
+    def define_change_status_method(status_name)
```

Comment:
Also, calling this method `define_transition` seems to be a better idea. Otherwise it's too easy to confuse `define_change_status_method` and `define_check_status_method`, both for me, for you ina  couple of months, and for your editor's autocompletion.